<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>H3K27me3 hESC Epigenomics data</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">hesc-epigenomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/elsasserlab/hesc-epigenomics">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">H3K27me3 hESC Epigenomics data</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>hesc-epigenomics/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomelsasserlabhescepigenomicstreed7e28c0a0c5d58236680cbd597f89ce2c08ffc6dtargetblankd7e28c0a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/elsasserlab/hesc-epigenomics/tree/d7e28c0a0c5d58236680cbd597f89ce2c08ffc6d" target="_blank">d7e28c0</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomelsasserlabhescepigenomicstreed7e28c0a0c5d58236680cbd597f89ce2c08ffc6dtargetblankd7e28c0a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/elsasserlab/hesc-epigenomics/tree/d7e28c0a0c5d58236680cbd597f89ce2c08ffc6d" target="_blank">d7e28c0</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/bed/
    Ignored:    data/bw
    Ignored:    data/igv/
    Ignored:    data/liftover/
    Ignored:    data/peaks
    Ignored:    data/rnaseq/
    Ignored:    figures_data/

Untracked files:
    Untracked:  data/meta/Kumar_2020_bins_panels_design.csv
    Untracked:  data/meta/Kumar_2020_master_bins_10kb_table_final_raw.tar.gz
    Untracked:  data/meta/Kumar_2020_master_bins_10kb_table_final_raw.tsv
    Untracked:  data/meta/Kumar_2020_master_bins_10kb_table_final_raw.zip
    Untracked:  data/meta/Kumar_2020_master_bins_10kb_table_raw (copy).tsv
    Untracked:  data/meta/Kumar_2020_master_bins_10kb_table_raw.tsv
    Untracked:  data/meta/Kumar_2020_master_bins_5kb_table_raw.tsv
    Untracked:  data/meta/Kumar_2020_master_bins_5kb_table_raw.zip
    Untracked:  data/meta/Kumar_2020_master_bins_5kb_table_replicates_only.tsv
    Untracked:  data/meta/Kumar_2020_master_bins_5kb_table_shrunk.tsv
    Untracked:  data/meta/Kumar_2020_master_bins_5kb_table_shrunk.zip
    Untracked:  data/meta/Kumar_2020_master_gene_table.zip
    Untracked:  data/meta/Kumar_2020_master_gene_table_rnaseq_shrunk.tsv
    Untracked:  data/meta/Kumar_2020_master_gene_table_rnaseq_shrunk_annotated.tsv
    Untracked:  data/meta/Kumar_2020_master_gene_table_rnaseq_shrunk_plus_annotations.tsv
    Untracked:  data/meta/Kumar_2020_master_gene_table_rnaseq_shrunk_plus_annotations_2.tsv
    Untracked:  data/meta/Kumar_2020_promoters_panels_design.csv
    Untracked:  data/meta/gene_names_bivalent.tsv
    Untracked:  data/other/Bock_2011/
    Untracked:  data/other/Cinkornpumin_2020/
    Untracked:  data/other/Lanner/
    Untracked:  data/other/Messmer_2019/Messmer_CellRep2019_TableS1_NaivePrimed_Diff.csv
    Untracked:  data/other/Messmer_2019/Messmer_CellRep2019_TableS1_NaivePrimed_Diff.xls
    Untracked:  data/other/Messmer_2019/Messmer_CellRep2019_TableS2_IntermediatePop.csv
    Untracked:  data/other/Messmer_2019/Messmer_CellRep2019_TableS2_IntermediatePop.xls
    Untracked:  data/other/Messmer_2019/Messmer_CellRep2019_TableS3_GermLayerMarkers.csv
    Untracked:  data/other/Messmer_2019/Messmer_CellRep2019_TableS3_GermLayerMarkers.xls
    Untracked:  data/other/Messmer_2019/Messmer_CellRep2019_TableS4_LineageEpiMarkers.csv
    Untracked:  data/other/Messmer_2019/Messmer_CellRep2019_TableS4_LineageEpiMarkers.xls
    Untracked:  output/Kumar_2020_H3K27m3_Ni_higher_than_Pr_tss.bed
    Untracked:  output/Kumar_2020_H3K27m3_Pr_higher_than_Ni_tss.bed
    Untracked:  output/Kumar_2020_H3K27m3_always_up_tss.bed
    Untracked:  output/fig2_chr_means_boxplot.svg
    Untracked:  output/fig3_violin_rnaseq_ratios_naive_higher_than_pr.svg
    Untracked:  output/fig3_violin_rnaseq_ratios_primed_higher_than_ni.svg
    Untracked:  output/fig4_lineage_makers_new.png
    Untracked:  output/fig4_lineage_makers_new.svg
    Untracked:  output/fig4_lineage_markers_new.png
    Untracked:  output/fig4_lineage_markers_new.svg
    Untracked:  output/fig4_lineage_markers_new_cluster.png
    Untracked:  output/fig4_lineage_markers_new_cluster.svg
    Untracked:  output/fig4_lineage_markers_new_k4.png
    Untracked:  output/fig4_lineage_markers_new_k4.svg
    Untracked:  output/fig4_messmer_top25.png
    Untracked:  output/fig4_messmer_top25.svg
    Untracked:  output/fig4_messmer_top25_down.png
    Untracked:  output/fig4_messmer_top25_down.svg
    Untracked:  output/fig4_messmer_top25_up.png
    Untracked:  output/fig4_messmer_top25_up.svg
    Untracked:  output/fig4_messmer_top50_down.png
    Untracked:  output/fig4_messmer_top50_down.svg
    Untracked:  output/fig4_messmer_top50_up.png
    Untracked:  output/fig4_messmer_top50_up.svg
    Untracked:  output/fig4_pca_all_plus_dong_data.svg
    Untracked:  output/fig4_pca_only_our_samples.svg
    Untracked:  output/fig4_pca_only_our_samples_labeled.svg
    Untracked:  output/fig4_pluripotency_markers_new.png
    Untracked:  output/fig4_pluripotency_markers_new.svg
    Untracked:  output/fig4_pluripotency_markers_new_cluster.png
    Untracked:  output/fig4_pluripotency_markers_new_cluster.svg
    Untracked:  output/panel_figures/
    Untracked:  output/promoter_panels_final/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/rnaseq_comparison.Rmd
    Modified:   code/heatmap_panels.R
    Modified:   code/heatmaply_functions.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/elsasserlab/hesc-epigenomics/blob/d7e28c0a0c5d58236680cbd597f89ce2c08ffc6d/analysis/index.Rmd" target="_blank">d7e28c0</a>
</td>
<td>
C. Navarro
</td>
<td>
2021-07-09
</td>
<td>
wflow_publish(“./analysis/index.Rmd”, verbose = T)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/elsasserlab/hesc-epigenomics/ce48d0f6941dff7fb8bdb50e65085b1b5c8ca0e2/docs/index.html" target="_blank">ce48d0f</a>
</td>
<td>
C. Navarro
</td>
<td>
2021-07-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/elsasserlab/hesc-epigenomics/blob/2d5c9aa6b7b50d3c7ea5eeff8c62289496e9258d/analysis/index.Rmd" target="_blank">2d5c9aa</a>
</td>
<td>
C. Navarro
</td>
<td>
2021-07-08
</td>
<td>
wflow_publish(“./analysis/index.Rmd”, verbose = T)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/elsasserlab/hesc-epigenomics/9e8b1a9327a1b3b4c38677f4bb346f610b402aa3/docs/index.html" target="_blank">9e8b1a9</a>
</td>
<td>
cnluzon
</td>
<td>
2021-05-26
</td>
<td>
Metadata folder added
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/elsasserlab/hesc-epigenomics/blob/0f37941ef6bcfbb3dc65c34a3452c778e729592b/analysis/index.Rmd" target="_blank">0f37941</a>
</td>
<td>
cnluzon
</td>
<td>
2021-03-12
</td>
<td>
Fig1 fresh
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/elsasserlab/hesc-epigenomics/0f37941ef6bcfbb3dc65c34a3452c778e729592b/docs/index.html" target="_blank">0f37941</a>
</td>
<td>
cnluzon
</td>
<td>
2021-03-12
</td>
<td>
Fig1 fresh
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/elsasserlab/hesc-epigenomics/blob/ef20c53e9c315fe47688ac95ea756ece6f380bd7/analysis/index.Rmd" target="_blank">ef20c53</a>
</td>
<td>
cnluzon
</td>
<td>
2021-03-11
</td>
<td>
Annotations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/elsasserlab/hesc-epigenomics/ef20c53e9c315fe47688ac95ea756ece6f380bd7/docs/index.html" target="_blank">ef20c53</a>
</td>
<td>
cnluzon
</td>
<td>
2021-03-11
</td>
<td>
Annotations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/elsasserlab/hesc-epigenomics/aa434ad5eed28f8352e090c4380b4b019d763da6/docs/index.html" target="_blank">aa434ad</a>
</td>
<td>
cnluzon
</td>
<td>
2021-02-08
</td>
<td>
Per chromosome density report
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/elsasserlab/hesc-epigenomics/blob/eb488560703d8e610082c71cdf7ff5a180d1204d/analysis/index.Rmd" target="_blank">eb48856</a>
</td>
<td>
cnluzon
</td>
<td>
2021-02-08
</td>
<td>
wflow_publish(“analysis/”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/elsasserlab/hesc-epigenomics/0437358f78995838585fb14638558bf40fc16ec9/docs/index.html" target="_blank">0437358</a>
</td>
<td>
cnluzon
</td>
<td>
2021-02-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/elsasserlab/hesc-epigenomics/blob/8942fdf8fa7c37484de8927ded480b2d812a6d3c/analysis/index.Rmd" target="_blank">8942fdf</a>
</td>
<td>
cnluzon
</td>
<td>
2021-02-07
</td>
<td>
Index page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/elsasserlab/hesc-epigenomics/8942fdf8fa7c37484de8927ded480b2d812a6d3c/docs/index.html" target="_blank">8942fdf</a>
</td>
<td>
cnluzon
</td>
<td>
2021-02-07
</td>
<td>
Index page
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/elsasserlab/hesc-epigenomics/blob/0481b52b31f8f9a2a188db336b93b2c5ea3b5f68/analysis/index.Rmd" target="_blank">0481b52</a>
</td>
<td>
cnluzon
</td>
<td>
2021-02-05
</td>
<td>
RNA seq first analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/elsasserlab/hesc-epigenomics/0481b52b31f8f9a2a188db336b93b2c5ea3b5f68/docs/index.html" target="_blank">0481b52</a>
</td>
<td>
cnluzon
</td>
<td>
2021-02-05
</td>
<td>
RNA seq first analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/elsasserlab/hesc-epigenomics/blob/87ff15f7aa5937570b780d7991ec3097562de76a/analysis/index.Rmd" target="_blank">87ff15f</a>
</td>
<td>
cnluzon
</td>
<td>
2021-02-05
</td>
<td>
Renamed read counts file
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/elsasserlab/hesc-epigenomics/87ff15f7aa5937570b780d7991ec3097562de76a/docs/index.html" target="_blank">87ff15f</a>
</td>
<td>
cnluzon
</td>
<td>
2021-02-05
</td>
<td>
Renamed read counts file
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/elsasserlab/hesc-epigenomics/7cafaa1ab93e11ca9e6ec93f4a42da8a430a368a/docs/index.html" target="_blank">7cafaa1</a>
</td>
<td>
cnluzon
</td>
<td>
2021-02-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/elsasserlab/hesc-epigenomics/blob/930c01545575b0deb8ce2637cc8c5ef35f5d6804/analysis/index.Rmd" target="_blank">930c015</a>
</td>
<td>
cnluzon
</td>
<td>
2021-02-03
</td>
<td>
wflow_publish(“analysis/”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/elsasserlab/hesc-epigenomics/331f00c452e487cde6e538c5e62e43bd1bd61942/docs/index.html" target="_blank">331f00c</a>
</td>
<td>
cnluzon
</td>
<td>
2021-02-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/elsasserlab/hesc-epigenomics/blob/04408d2a7f195ebd3bef939b2cf70f3f0db02f25/analysis/index.Rmd" target="_blank">04408d2</a>
</td>
<td>
cnluzon
</td>
<td>
2021-02-03
</td>
<td>
Global read counts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/elsasserlab/hesc-epigenomics/blob/897ecf574e664028f1303659638001e43c1e6b17/analysis/index.Rmd" target="_blank">897ecf5</a>
</td>
<td>
cnluzon
</td>
<td>
2021-02-02
</td>
<td>
wflow_git_commit(all = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/elsasserlab/hesc-epigenomics/blob/407fa2068fedf46cdab878a67773e8343391da39/analysis/index.Rmd" target="_blank">407fa20</a>
</td>
<td>
cnluzon
</td>
<td>
2021-02-02
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Supplementary code and analysis for publication.</p>
<div id="publication-figures" class="section level1">
<h1>Publication figures</h1>
<ul>
<li><a href="fig_01_quantitative_chip.html">Figure 1</a>. High levels of H3K27m3 and H2AUb broadly cover the naïve pluripotent genome.</li>
<li><a href="fig_02_x_chromosome.html">Figure 2</a>. X chromosome in naïve hESCs accumulate exceptional amounts of H3K27m3 without a globally repressive function.</li>
<li><a href="fig_03_h3k27m3_groups.html">Figure 3</a>. H3K27m3 is adaptive to gene expression changes between naïve and primed pluripotent state and contributes to repression of non-state specific genes.</li>
<li><a href="fig_04_intermediate.html">Figure 4</a>. Loss of H3K27m3 in naïve hESCs derepresses a set of genes that drive hESCs towards a rare naturally ocurring intermediate pluripotent state.</li>
<li><a href="fig_04_intermediate.html">Figure 5</a>. Loss of H3K27m3 in naïve hESCs activates trophectoderm and placental gene expression programs.</li>
</ul>
</div>
<div id="supplementary-data-and-analyses" class="section level1">
<h1>Supplementary data and analyses</h1>
<ul>
<li><a href="master_gene_table.html">Master gene table</a>. Notebook with the code that generates the master gene table.</li>
<li><a href="master_bins_table.html">Master bins table</a>. Notebook with the code that generates the master bins table.</li>
</ul>
</div>
<div id="main-data-files" class="section level1">
<h1>Main data files</h1>
<p>A large proportion of the figures shown in this publication can be drawn directly from values in the main tables:</p>
<ul>
<li><a href="../data/meta/Kumar_2020_master_gene_table_rnaseq_shrunk_annotated.zip">Gene table</a>. Contains histone mark values across 5kb regions around gene TSS in the hg38 genome, RNA-seq plus statistics on differentially decorated promoters and differentially expressed genes across conditions. differentially decorated regions.</li>
<li><a href="../data/meta/Kumar_2020_master_bins_10kb_table_final_raw.zip">Bins table</a>. Gathers histone mark values across 10kb windows in the hg38 genome, plus statistics on differentially decorated regions across conditions.</li>
</ul>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
