---
title: "H3K27me3 hESC Epigenomics data"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

Supplementary code and analysis for publication.

# Publication figures

* [Figure 1](fig_01_quantitative_chip.html). High levels of H3K27m3
and H2AUb broadly cover the naïve pluripotent genome.
* [Figure 2](fig_02_x_chromosome.html). X chromosome in naïve hESCs 
accumulate exceptional amounts of H3K27m3 without a globally repressive function.
* [Figure 3](fig_03_h3k27m3_groups.html). H3K27m3 is adaptive to gene expression
changes between naïve and primed pluripotent state and contributes to repression
of non-state specific genes.
* [Figure 4](fig_04_intermediate.html). Loss of H3K27m3 in naïve hESCs derepresses
a set of genes that drive hESCs towards a rare naturally ocurring intermediate
pluripotent state.
* [Figure 5](fig_04_intermediate.html). Loss of H3K27m3 in naïve hESCs 
activates trophectoderm and placental gene expression programs.

# Supplementary data and analyses

* [Master gene table](master_gene_table.html). Notebook with the code that 
generates the master gene table. 
* [Master bins table](master_bins_table.html). Notebook with the code that 
generates the master bins table.

# Main data files

A large proportion of the figures shown in this publication can be drawn directly
from values in the main tables:

* [Gene table](../data/meta/Kumar_2020_master_gene_table_rnaseq_shrunk_plus_annotations.zip).
 Contains histone mark values across 5kb regions around gene TSS in the hg38 genome, RNA-seq
 plus statistics on differentially decorated promoters and differentially expressed
 genes across conditions.
differentially decorated regions.
* [Bins table](../data/meta/Kumar_2020_master_bins_10kb_table_final_raw.zip). Gathers
histone mark values across 10kb windows in the hg38 genome, plus statistics on
differentially decorated regions across conditions.


